### PR TITLE
Refactor: CampaignModal to use DateRangePicker for campaign dates

### LINKED ISSUE #5176: Refactor CampaignModal to use DateRangePicker for campaign dates
## Overview
Replace startAt/endAt date selection in CampaignModal.tsx with DateRangePicker component. This removes duplicate state management and provides consistent date range UX.

## Related Issues
- Depends on: #5173

## Files to Update
- `src/screens/OrganizationFundCampaign/modal/CampaignModal.tsx` - Replace startAt/endAt with DateRangePicker
- `src/screens/OrganizationFundCampaign/modal/CampaignModal.spec.tsx` - Update tests

## Dependencies
- Blocked by: # (DateRangePicker component creation)

## Acceptance Criteria
- [ ] DateRangePicker component imported and used
- [ ] startAt/endAt state replaced with dateRange state
- [ ] Duplicate date selection UI removed
- [ ] GraphQL mutation variables extract dates correctly
- [ ] Built-in DateRangePicker validation leveraged
- [ ] All tests updated and passing
- [ ] Campaign creation and editing work correctly

## Testing Checklist
- [ ] Campaign creation with date range works
- [ ] Campaign editing preserves dates correctly
- [ ] Date validation prevents invalid ranges
- [ ] GraphQL mutations receive correct ISO date strings
- [ ] Form submission handles dates properly

## Estimated Effort
6 hours

## Implementation Notes
- Replace `campaignStartDate` and `campaignEndDate` in formState with unified `dateRange`
- Extract dates when calling createCampaign/updateCampaign mutations
- Maintain existing `dayjs().toISOString()` formatting

### PR DESCRIPTION
**Pull request by s3arthak**
**What kind of change does this PR introduce?**  
Refactoring

**Issue Number:**  
Fixes #5176  
Parent: #5173

**Snapshots/Videos:**  
N/A

**If relevant, did you update the documentation?**  
N/A

---

## Summary

This PR refactors the `CampaignModal` to replace the existing `startAt` and `endAt` date handling logic with the reusable `DateRangePicker` component.  

The change removes duplicate date state management, ensures consistent date range UX across the application, and leverages the built-in validation logic of `DateRangePicker`. It also simplifies GraphQL mutation handling by extracting campaign dates directly from a unified date range state.

---

**Does this PR introduce a breaking change?**  
No

---

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI  
- [x] I have implemented or provided justification for each non-critical suggestion  
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented  

### Test Coverage
- [x] I have written tests for all new changes/features  
- [x] I have verified that test coverage meets or exceeds 95%  
- [x] I have run the test suite locally and all tests pass  

---

**Other information**

- Replaced individual `startAt` / `endAt` inputs with `DateRangePicker`
- Unified date handling using a single `dateRange` state
- Ensured mutations still send ISO-formatted date strings
- Updated tests to mock and validate `DateRangePicker` behavior
- Verified campaign creation and editing flows

---

**Have you read the contributing guide?**  
Yes

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected a spelling error in campaign-related error text.

* **New Features**
  * Added comprehensive campaign validation messages across English, Spanish, French, Hindi, and Chinese.
  * Improved date-range selection with stricter validation to prevent invalid or reversed dates.
  * Enhanced notifications for missing campaigns, invalid dates, required fields, and submission states.

* **Tests**
  * Expanded test coverage for campaign create/edit flows, date handling, and validation scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by s3arthak** [comment]
@coderabbitai review

**Comment by s3arthak** [comment]
> > > > @coderabbitai full review and reevaluate all resolved items and include the following points:
> > > > 
> > > > 1. **Test Coverage & Quality Validation:**
> > > >    
> > > >    1. Validate whether any updated test files:
> > > >       
> > > >       1. Cover all use cases including edge cases, error scenarios, and success paths
> > > >       2. Have valid, properly structured tests following Jest and React Testing Library best practices
> > > >       3. Provide 100% test code coverage for all modified/added functions, components, and branches
> > > >       4. Include proper mocking of GraphQL queries/mutations, external dependencies, and API calls
> > > >       5. Test both user interactions and state changes comprehensively
> > > >       6. Require any additional tests for uncovered scenarios
> > > >    2. For each modified file with logic or UI, verify corresponding test files exist and are updated
> > > >    3. Check that test files follow naming conventions (*.test.tsx, *.test.ts, *.spec.tsx, *.spec.ts)
> > > >    4. Validate that tests don't include disabled tests (it.skip, describe.skip) unless explicitly documented
> > > >    5. Ensure mock data cleanup is performed (as required by pre-commit hooks)
> > > > 2. **Code Quality & Repository Standards:**
> > > >    
> > > >    1. Verify TypeScript compliance:
> > > >       
> > > >       1. No use of `any` type without explicit justification
> > > >       2. Proper interface/type definitions for all props, state, and function parameters
> > > >       3. No TypeScript errors or warnings
> > > >    2. Validate GraphQL implementation:
> > > >       
> > > >       1. Queries/mutations are properly typed and follow existing patterns in src/GraphQl/
> > > >       2. No duplicate or redundant queries (check for pagination inconsistencies)
> > > >       3. Error handling is implemented for all GraphQL operations
> > > >    3. Check React component best practices:
> > > >       
> > > >       1. Functional components with proper hooks usage
> > > >       2. Material-UI component patterns are followed consistently
> > > >       3. No prop drilling; proper use of context when needed
> > > >       4. Proper cleanup in useEffect hooks
> > > >    4. Verify internationalization (i18n):
> > > >       
> > > >       1. All user-facing strings use i18next translation keys
> > > >       2. New translation keys are added to locale files
> > > >       3. No hardcoded English text in UI components
> > > >    5. Validate accessibility:
> > > >       
> > > >       1. Proper ARIA labels and roles
> > > >       2. Keyboard navigation support
> > > >       3. Screen reader compatibility
> > > > 3. **Documentation & Code Comments:**
> > > >    
> > > >    1. Verify auto-generated documentation is updated (as enforced by Check-AutoDocs job)
> > > >    2. Check for adequate JSDoc comments on public functions and complex logic
> > > >    3. Ensure README.md or related docs are updated if new features are added
> > > >    4. Validate that Table of Contents is updated (as required by pre-commit hooks)
> > > > 4. **Identify Issues:**
> > > >    
> > > >    1. Lines of code that are missing tests (list specific line numbers)
> > > >    2. Any unnecessary files that have been submitted (build artifacts, node_modules, .env files, IDE configs)
> > > >    3. Unused imports or dead code (as checked by knip)
> > > >    4. Console.log statements or debug code left in production files
> > > >    5. Files that should be excluded from the PR scope
> > > > 5. **PR Requirements Validation:**
> > > >    
> > > >    1. Verify the PR meets the requirements of the linked issue:
> > > >       
> > > >       1. All acceptance criteria are addressed
> > > >       2. No out-of-scope changes are included
> > > >    2. Validate compliance with all previously suggested improvements and comments:
> > > >       
> > > >       1. All review comments have been addressed
> > > >       2. Requested changes have been implemented correctly
> > > >    3. Check PR metadata:
> > > >       
> > > >       1. Proper title and description
> > > >       2. Linked to appropriate issue(s)
> > > >       3. Target branch is correct (typically `develop`)
> > > >    4. Verify no sensitive information is exposed (as checked by Check-Sensitive-Files job)
> > > > 6. **CI/CD Pipeline Compatibility:**
> > > >    
> > > >    1. Validate that code changes won't break any of the 17+ CI jobs
> > > >    2. Ensure pre-commit hook requirements are met:
> > > >       
> > > >       1. Code formatting (Prettier)
> > > >       2. Linting (ESLint)
> > > >       3. Type checking (TypeScript)
> > > >    3. Check that changes don't introduce performance regressions for the 12-shard test execution
> > > > 7. **Security & Best Practices:**
> > > >    
> > > >    1. No security vulnerabilities introduced
> > > >    2. No exposed API keys, tokens, or credentials
> > > >    3. Proper input validation and sanitization
> > > >    4. No SQL injection or XSS vulnerabilities in any data handling
> > > > 8. **Review Decision:**
> > > >    
> > > >    * If the submitted code requires **any changes** (even trivial ones such as missing tests, typing issues, documentation gaps, or minor improvements):
> > > >      
> > > >      1. Provide detailed, actionable feedback for each issue
> > > >      2. **Explicitly change the PR status to CHANGES_REQUESTED using:** `gh pr review <PR_NUMBER> --request-changes -b "<summary of required changes>"`
> > > >    * If **no changes are required** and all criteria above are satisfied:
> > > >      
> > > >      1. **Explicitly approve the PR using:** `gh pr review <PR_NUMBER> --approve -b "All validation checks passed. LGTM! ✅"`
> > > > 9. **Focus Areas:**
> > > >    
> > > >    * Prioritize substantive issues over minor stylistic suggestions
> > > >    * Focus on issues that would cause CI pipeline failures or production bugs
> > > >    * Highlight any deviations from established codebase patterns

**Comment by s3arthak** [comment]
@palisadoes ready for the review

File: src/screens/AdminPortal/OrganizationFundCampaign/modal/CampaignModal.tsx [hunk 1]
```diff
@@ -12,18 +18,17 @@ import {
...
   CREATE_CAMPAIGN_MUTATION,
   UPDATE_CAMPAIGN_MUTATION,
 } from 'GraphQl/Mutations/CampaignMutation';
 import { NotificationToast } from 'components/NotificationToast/NotificationToast';
-import BaseModal from 'shared-components/BaseModal/BaseModal';
-import DatePicker from 'shared-components/DatePicker';
-import { currencyOptions, currencySymbols } from 'utils/currency';
-import { errorHandler } from 'utils/errorHandler';
-import type { InterfaceCampaignInfo } from 'utils/interfaces';
-import styles from './CampaignModal.module.css';
```
Comment (Piyush-Goenka) [inline src/screens/AdminPortal/OrganizationFundCampaign/modal/CampaignModal.tsx:26]: Why `import styles from './CampaignModal.module.css` this has been replaced with the global style sheet ?

**Comment by Piyush-Goenka** [comment]
@palisadoes 
Why this import styles from local style sheet in this PR had been changed to global style sheet ?

**Comment by s3arthak** [comment]
@Piyush-Goenka Thanks for flagging this , 
I will address it by creating a separate issue and fix it there.

**Comment by palisadoes** [comment]
> @palisadoes Why this import styles from local style sheet in this PR had been changed to global style sheet ?

@Piyush-Goenka 

Please open an issue for this

**Comment by Piyush-Goenka** [comment]
@palisadoes
No need to open the issue.
Fixing it in my upcoming PR.

**Comment by Piyush-Goenka** [comment]
@s3arthak 
Don't raise a separate PR for this.
Currently I am working on the refactoring.
So, will fix it there.
